pkg/controller/migmigration/migmigration_controller.go

```diff
@@ -149,11 +163,53 @@ func (r *ReconcileMigMigration) Reconcile(request reconcile.Request) (reconcile.
 		}
 	}
 
-	// Delayed requeue to support polling.
-	if requeue {
-		delay := time.Second * 5
+	// Requeue
+	if requeueAfter > 0 {
+		delay := time.Second * time.Duration(requeueAfter)
 		return reconcile.Result{RequeueAfter: delay}, nil
 	}
 
 	return reconcile.Result{}, nil
 }
+
+// Determine if a migration should be postponed.
+// Migrations run serially ordered by created timestamp and grouped
+// with stage migrations followed by final migrations. A migration is
+// postponed when not in the desired order.
+// Returns: a delay in seconds to postpone, else 0 (not postponed).
+func (r *ReconcileMigMigration) postpone(migration *migapi.MigMigration) (int, error) {
+	plan, err := migration.GetPlan(r)
+	if err != nil {
+		return 0, err
+	}
+	migrations, err := plan.ListMigrations(r)
+	if err != nil {
+		return 0, err
+	}
+	// Pending migrations.
+	pending := []types.UID{}
+	for _, m := range migrations {
+		if !m.Status.MigrationCompleted {
+			pending = append(pending, m.UID)
+		}
+	}
+	// This migration is next.
+	if pending[0] == migration.UID {
+		return 0, nil
+	}
+	// Postpone
+	for position, uid := range pending {
+		migration.Status.SetCondition(migapi.Condition{
+			Type:     Postponed,
+			Status:   True,
+			Category: Critical,
+			Message:  PostponedMessage,
+		})
```

Comment:
I'm not quite following the workflow here. It looks like we're setting the Postponed condition for this migration each time through the loop to the same thing, and then when we find the position of this migration, we use it to set the requeue time. Should this statement be outside the loop? Should it always run or only if there's at least one pending migration?

Reply:
Yes, meant to put it outside the loop.  Fixed.